ApplicationUnitTests/Commands/Identities/CreateIdentityHandlerTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿using Identity_Application.Commands.Identities;
+using Identity_Application.Interfaces.Authentication;
+using Identity_Application.Interfaces.Repository;
+using Identity_Application.Models.AppSettingsModels;
+using Identity_Application.Models.BaseEntitiesModels;
+using Identity_Domain.Entities.Additional;
+using Identity_Domain.Entities.Base;
+using Microsoft.Extensions.Options;
+using Moq;
+
+namespace ApplicationUnitTests.Commands.Identities;
+
+public class CreateIdentityHandlerTests
+{
+    private readonly Mock<IGenericRepository<Identity>> _mockIdentityRepository;
+    private readonly Mock<IPasswordHasherService> _mockPasswordHasher;
+    private readonly Mock<IOptions<PasswordHashSettings>> _mockConfig;
+    private readonly CreateIdentityHandler _handler;
+
+    public CreateIdentityHandlerTests()
+    {
+        _mockIdentityRepository = new Mock<IGenericRepository<Identity>>();
+        _mockPasswordHasher = new Mock<IPasswordHasherService>();
+        _mockConfig = new Mock<IOptions<PasswordHashSettings>>();
+        _handler = new CreateIdentityHandler(_mockIdentityRepository.Object,
+                                             _mockPasswordHasher.Object,
+                                             _mockConfig.Object);
+    }
+
+    [Fact]
+    public async Task Handle_ValidCommand_ShouldCreateIdentity()
+    {
+        // Arrange
+        var vm = new IdentityVM
+        {
+            Username = "NewUser",
+            Email = "newuser@example.com",
+            Password = "password123",
+            ClaimsIds = new int?[] { 1, 2 },
+            RolesIds = new int?[] { 1 }
+        };
+
+        var command = new CreateIdentityCommand(vm);
+
+        var salt = "random_salt";
+        var hash = "hashed_password";
+
+        var expectedIdentity = new Identity
+        {
+            Username = "NewUser",
+            Email = "newuser@example.com",
+            PasswordSalt = salt,
+            PasswordHash = hash
+        };
+
+        foreach (var claim in command.IdentityVM.ClaimsIds)
```

Comment:
just comment - I'm not big fan of loops in unit tests

Reply:
Yeah I remember that, but it is necessary, but too little and too important to provide that in some helper class/method